xformers/components/attention/csrc/cpu/attention.cpp

```diff
@@ -28,10 +28,12 @@ scalar_t max(scalar_t* buf) {
 template <typename scalar_t>
 void attention_kernel(
     at::TensorAccessor<scalar_t, 3> output,
+    at::TensorAccessor<scalar_t, 2> logsumexp,
     at::TensorAccessor<scalar_t, 3> query,
     at::TensorAccessor<scalar_t, 3> key,
     at::TensorAccessor<scalar_t, 3> value,
-    at::TensorAccessor<scalar_t, 3> buffer //,
+    at::TensorAccessor<scalar_t, 3> buffer,
+    bool compute_logsumexp
     // at::TensorAccessor<int64_t, 2> mask
```

Comment:
ahah, next step is to make it sparse ? :D (not that much of a joke..)

Reply:
yes, integrating sparsity is in the plans :-) But before, I'll look into the `K > 32` case